tools/autograd/derivatives.yaml

```diff
@@ -187,9 +187,9 @@
   tensor2: handle_r_to_c(tensor2.scalar_type(), grad * (tensor1 * value).conj())
 
 - name: addmm(Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
-  self: "grad.is_mkldnn() ? grad.to_dense() : maybe_multiply(grad, beta.conj())"
-  mat1: "grad.is_mkldnn() ? mkldnn_linear_backward_input(mat1.sizes(), grad, mat2.t()) : mm_mat1_backward(grad, mat2, mat1.sizes(), mat1.strides(), alpha)"
-  mat2: "grad.is_mkldnn() ? (std::get<0>(mkldnn_linear_backward_weights(grad, mat1, mat2.t(), true))).t() : mm_mat2_backward(grad, mat1, mat2.sizes(), mat2.strides(), alpha)"
+  self: maybe_multiply(grad, beta.conj())
```

Comment:
wait, how is addmm for mkldnn tensors supposed to work now? These lines were introduced by #49453, and they are needed. 
@XiaobingSuper why are there no tests that would catch this?

Reply:
I thought the other PR is trying to enable it for linear, and has to go through admm since it is how linear was represented in JIT.
So that part is done via autograd below:
```
- name: mkldnn_linear(Tensor self, Tensor weight, Tensor? bias=None) -> Tensor
  self, weight, bias: mkldnn_linear_backward(self, grad, weight, grad_input_mask)
```

I don't think the admm part would be trouble some in our context. So we could try to put them back if it's intended to be kept.

Reply:
I don't see this PR removing `mkldnn_addmm_wraper` from native_functions.yaml, which means that addmm backward is needed.

Reply:
Ah, sorry, it's actually removed, fine then.

Reply:
Let me try to throw a test there and patch it.

Reply:
No, please don't bring mkldnn_addmm_wraper back, it's good that it's removed, without your PR addmm for mkldnn tensors was in pretty bad shape, but now luckily we don't need to care about it.

Reply:
I notice that we did indeed remove `mkldnn_addmm_wraper` and from the comment it's intended

https://github.com/pytorch/pytorch/pull/51613/files/919a29a171b6c26bc70e8b88511bfbe148d5d978#diff-172c536bf0e2a56a98b3b07941391e3630fff843b5707a606f1fb5eafac2c35cL16

I'm trying a build to revert the patch and only apply the autograd rule for `mkldnn_linear` (so we can keep mkldnn_addmm). Please let me know if this is needed. @XiaobingSuper

Reply:
No, we don't want to keep mkldnn_addmm

Reply:
Got it.

I actually got my local build to work and `mkldnn_addmm_wraper` is giving me wrong results.
I traced it down to aten::mkldnn_linear. it can't handle non-contiguous weight. Since we are not using the wrapper which does that transpose, it's not an issue any more. CC'ing @XiaobingSuper for future reference.

Reply:
**mkldnn_addmm_wraper** is for call in mkldnn path for some linear case(input' s dim is two, and has bias, linear will call in addmm), if  aten linear ported, we can removed it.

Reply:
> wait, how is addmm for mkldnn tensors supposed to work now? These lines were introduced by #49453, and they are needed.
> @XiaobingSuper why are there no tests that would catch this?

#49453 is for catch linear case when linear call in addmm , there just has test linear test case.

Reply:
FYI #51713